backpack-compose/src/main/kotlin/net/skyscanner/backpack/compose/navigationbar/TopNavBarState.kt

```diff
@@ -41,25 +39,46 @@ fun Modifier.nestedScroll(state: TopNavBarState): Modifier =
     nestedScroll(state.asInternalState().nestedScrollConnection)
 
 @Composable
+@OptIn(ExperimentalMaterial3Api::class)
 fun rememberTopAppBarState(initialStatus: TopNavBarStatus = TopNavBarStatus.Expanded): TopNavBarState {
-    val offsetRange = with(LocalDensity.current) {
-        (TopNavBarSizes.ExpandedHeight - TopNavBarSizes.CollapsedHeight).toPx()
+    val behaviour = TopAppBarDefaults.enterAlwaysScrollBehavior(
+        state = when (initialStatus) {
+            TopNavBarStatus.Expanded -> androidx.compose.material3.rememberTopAppBarState(
+                initialHeightOffset = 0f,
+                initialContentOffset = 0f,
+            )
+
+            TopNavBarStatus.Collapsed -> androidx.compose.material3.rememberTopAppBarState(
+                initialHeightOffset = -Float.MAX_VALUE,
+                initialContentOffset = -Float.MAX_VALUE,
+            )
+        },
+    )
+    return remember(behaviour) {
+        TopNavBarInternalState(behaviour)
     }
-    val flingBehavior = ScrollableDefaults.flingBehavior()
-    return rememberSaveable(
-        offsetRange, flingBehavior,
-        saver = TopNavBarStateImpl.saver(offsetRange, flingBehavior),
-        init = { TopNavBarStateImpl(initialStatus, offsetRange = offsetRange, flingBehavior = flingBehavior) },
+}
+
+@Composable
+@OptIn(ExperimentalMaterial3Api::class)
+internal fun rememberFixedTopAppBarState(): TopNavBarState {
+    val behaviour = TopAppBarDefaults.pinnedScrollBehavior(
+        canScroll = { false },
+        state = androidx.compose.material3.rememberTopAppBarState(
+            initialHeightOffset = -Float.MAX_VALUE,
+            initialContentOffset = -Float.MAX_VALUE,
+        ),
     )
+    return remember(behaviour) {
+        TopNavBarInternalState(behaviour)
+    }
 }
 
 @Stable
-internal interface TopNavBarInternalState : TopNavBarState {
-
-    val fraction: Float
-
-    val nestedScrollConnection: NestedScrollConnection
-}
+@OptIn(ExperimentalMaterial3Api::class)
```

Comment:
It was previously one internal interface + one internal implementation. Merged now.